### ISSUE TITLE
COLAB VERSION PROBLEM SOLUTION

### ISSUE THREAD

**Issue by userwatch** [matterport/Mask_RCNN#3041]
Hi everyone,  I used this repo in Colab. https://github.com/z-mahmud22/Mask-RCNN_TF2.14.0    If it doesn't work, let me know and I will share my own colab file. 

Best wishes,

**Comment by zikav1** [comment]
Have you been training on your own dataset? Me and my friend have been using that repo and have gotten some decent results, however the detection is giving some bad results and the mAP always returns 0.0...

**Comment by userwatch** [comment]
yes i tried own dataset, It didn't give any errors but the results are not very good. Also, training is slow, how can I fast it up?

**Comment by zikav1** [comment]
No the results are not best for us either, only decent depending on the configs we use. Don't know much about your specs but I guess you are running the training with a GPU?

**Comment by userwatch** [comment]
In Google Colab, I select the runtime as GPU. But I don't know if you use it or not.

https://github.com/userwatch/test/blob/main/OneClassCar.ipynb I added the model codes here

**Comment by zikav1** [comment]
You can check that by creating a cell and run the line "tf.test.is_gpu_available()" I think. Because you want to run tensorflow with the GPU

**Comment by userwatch** [comment]
![image](https://github.com/matterport/Mask_RCNN/assets/69357065/3f5ba244-03ea-4519-b6ff-e08e8dc01088)
showed this result.





![image](https://github.com/matterport/Mask_RCNN/assets/69357065/e154fcda-2bb0-4236-8b3b-0584c366b54c)
This is requirements.txt

**Comment by userwatch** [comment]
![image](https://github.com/matterport/Mask_RCNN/assets/69357065/e154fcda-2bb0-4236-8b3b-0584c366b54c)
This is requirements.txt

**Comment by userwatch** [comment]
![image](https://github.com/matterport/Mask_RCNN/assets/69357065/0aab1301-00de-4f60-898d-245131c6613c)

**Comment by userwatch** [comment]
!pip install tf-nightly  
![image](https://github.com/matterport/Mask_RCNN/assets/69357065/5ee8d083-d55e-4704-b583-2dedb6b9a3e6)
 I can see your name now

**Comment by userwatch** [comment]
![image](https://github.com/matterport/Mask_RCNN/assets/69357065/732f8655-a3ae-49d9-a8c4-385331d3d530)
 I encountered a problem with the version again:(

**Comment by userwatch** [comment]
Which version should it be? tf-nightly  Do you have any ideas? @zikav1 

https://stackoverflow.com/questions/76453565/error-installing-tensorflow-gpu-in-google-colab-subprocess-exited-with-error
I think there is no need to install it, just connecting to the GPU at runtime is enough.


gpu looks before installing requirements. 
 GPU:[] does not appear after this line of code (!pip install -r /content/maskrcnn/requirements.txt)

**Comment by userwatch** [comment]
https://stackoverflow.com/questions/76453565/error-installing-tensorflow-gpu-in-google-colab-subprocess-exited-with-error
I think there is no need to install it, just connecting to the GPU at runtime is enough.

**Comment by zikav1** [comment]
I'm not quite sure, we're not working inside of google colab. We're using a WSL ubuntu env and have installed CUDA and cuDNN to be able to run tensorflow with the GPU.

**Comment by userwatch** [comment]
This problem has been resolved. Thank you very much.

**Comment by userwatch** [comment]
How are your results?

**Comment by zikav1** [comment]
They are not that good. Don't know what is wrong. Something seems off with the validation and detection but can't figure out what that could be. And you?

**Comment by userwatch** [comment]
How many epochs and steps_per_epoch did you use?  In the example I follow step_per_epoch =1000
epoch=5  It detects in the testing part.  https://github.com/TannerGilbert/MaskRCNN-Object-Detection-and-Segmentation/blob/master/MaskRCNN%20Microcontroller%20Segmentation.ipynb     
That's my test.
![yanii](https://github.com/matterport/Mask_RCNN/assets/69357065/78748f84-70eb-4841-86ea-70fd1c974482)

I can share my data file with you. or you can also try the  microcontroller dataset here.


Note: Try testing with the best model in .h5 files.

**Comment by zikav1** [comment]
We have been trying with different optimizations. But think we ran something like 50 steps/epoch and 5000 epochs or so. Still getting some weird results when trying to predict. Yes that would be really nice if you could share that with me. 

And yes i'm a bit confused about what is being the model and the .h5 files. Is a single .h5 file defined as the "model"?

**Comment by zikav1** [comment]
Do you have discord or anything, would be easier to talk through there

**Comment by userwatch** [comment]
https://github.com/userwatch/data  I added the data here. Yes, I use discord,  my username is usertttwm

**Comment by zikav1** [comment]
Thanks, is it okay if I add you?

**Comment by userwatch** [comment]
Yes, you can add,
We help each other.

**Comment by zikav1** [comment]
Yes perfect! I added you
